.ci/compat/run.sh

```diff
@@ -0,0 +1,251 @@
+  meson setup "$1" -D prefix="/opt/$1"
+  meson install -C "$1"
+}
+
+build_autotools() {
+  autoreconf -fsi
+  ./configure --prefix="/opt/$1"
+  make -j"$(nproc)"
+  make install
+}
+
+build() {
+  local ref="$1"
+  header "Building tinc (ref $ref)"
+
+  git clone "$PWD" "$src/$ref" -b "compat-$ref"
+  pushd "$src/$ref"
+
+  if [[ -f meson.build ]]; then
+    build_meson "$ref"
+  else
+    build_autotools "$ref"
+  fi
+
+  popd
+  mkdir -p "/opt/$ref/var/run"
+}
+
+wait_network() {
+  local from="$1"
+  local to="$2"
+  local total=0
+
+  while ! ip netns exec "$from" ping -W1 -c1 "${addr[$to]}" >/dev/null; do
+    total=$((total + 1))
+
+    if [[ total -gt 60 ]]; then
+      echo >&2 "Network connection between $from and $to is not working"
+      exit 1
+    fi
+
+    echo >&2 "Network isn't ready yet..."
+    sleep 1
+  done
+}
+
+test_network() {
+  local from="$1"
+  local to="$2"
+
+  wait_network "$from" "$to"
+
+  header "Sending data between $from and $to"
+
+  ip netns exec "$from" \
+    iperf3 --time 1 --client "${addr[$to]}"
+}
+
+test_sign_verify() {
+  local signer="$1"
+  local verifier="$2"
+  local output="$etc/$signer/signed"
+
+  header "Test signature verification between $signer and $verifier"
+
+  "$signer" sign >"$output" <<<"text message for node $signer to sign"
+
+  for peer in "$signer" '*'; do
+    "$verifier" verify "$peer" "$output"
+    "$verifier" verify "$peer" <"$output"
+  done
+}
+
+test_node_status() {
+  local node="$1"
+
+  header "Checking node status for $node"
+
+  reachable="$("$node" dump reachable nodes | wc -l)"
+  echo >&2 "Node $node can reach $reachable nodes"
+
+  [[ $reachable == "$total_nodes" ]]
+
+  for peer in $nodes; do
+    echo >&2 -n "$node info $peer: "
+    "$node" info "$peer" | tee /dev/stderr | grep -E -q '(can reach itself|directly)'
+  done
+}
+
+latest() {
+  /opt/latest/sbin/tinc -c $etc/latest "$@"
+}
+
+tinc11() {
+  /opt/tinc11/sbin/tinc -c $etc/tinc11 "$@"
+}
+
+header 'Creating branches'
+
+for node in $nodes; do
+  git branch "compat-$node" "${refs[$node]}"
+  echo >&2 "    $node: $(git rev-parse "compat-$node")"
+done
+
+build tinc10
+build tinc11
+build latest
+
+header 'Initializing node from the latest commit'
+
+latest <<EOF
+  init latest
+  set Port 30000
+  set Interface latest
+  set Subnet ${addr[latest]}
+  set Address localhost
+  set Compression 0
+  set LogLevel 5
+EOF
+
+header 'Initializing node from the latest tag'
+
+tinc11 <<EOF
+  init tinc11
+  set Port 30001
+  set Interface tinc11
+  set Subnet ${addr[tinc11]}
+  set Address localhost
+  set Compression 3
+  set LogLevel 5
+EOF
+
+header 'Initializing node for tinc 1.0'
+
+latest -c $etc/tinc10 <<EOF
+  init tinc10
+  set Port 30002
+  set Interface tinc10
+  set Subnet ${addr[tinc10]}
+  set Address localhost
+  set Compression 10
+  set LogLevel 5
+EOF
+
+# Remove EC public key or newer nodes will complain about protocol downgrade
+sed -i 's#^Ed25519PublicKey##' $etc/tinc10/hosts/tinc10
```

Comment:
You could just add `del Ed25519PublicKey` to the commands you pass to the CLI. But more importantly, it might be better to initialize an 1.0 node the 1.0 way, by creating a `tinc.conf` and `hosts/tinc10` file with everything but the key, and then run `tincd -c $etc/tinc10 -K` to generate the key.

Reply:
The thought was to check that newer CLIs can create something that works even with old servers, but this makes more sense. Replaced.